zerver/tests/test_typing.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from zerver.lib.test_helpers import (
+    ZulipTestCase
+)
+
+class TypingPOSTTest(ZulipTestCase):
+    def test_typing_notification_to_self(self):
+        # type: () -> None
+        """
+        Sending typing notification to yourself
+        is successful.
+        """
+	email = "hamlet@zulip.com"
+        result = self.client_post("/api/v1/typing", {"type": "private",
+							"to": "hamlet@zulip.com"},
+					**self.api_auth(email))
+	self.assert_json_success(result)
+    
+    def test_typing_notification_to_another_user(self):
+        # type: () -> None
+        """
+        Sending typing notification to another user
+        is successful.
+        """
+        sender = "hamlet@zulip.com"
+        reciever = "othelo@zulip.com"
+        result = self.client_post("/api/v1/typing", {"type": "private",
+                                                    "to": reciever},
+                                        **self.api_auth(sender)
+        self.assert_json_success(result)
+
+    def test_typing_notification_without_reciever(self):
+        #type: () -> None
+        """
+        Sending typing notification without receiver fails
+        """
+        sender = "hamlet@zulip.com"
+        result = self.client_post("/api/v1/typing", {"type": "private"},
+                                        **self.api_auth(sender)
+        self.assert_json_error(result, "Notification reciever is required")
+
+    def test_typing_notification_without_type(self):
+        #type: () -> None
+        """
+        Sending typing notification without type
+        """
+        sender = "hamlet@zulip.com"
+        reciever = "othelo@zulip.com"
+        result = self.client_post("/api/v1/typing", {"to": reciever},
+                                        **self.api_auth(sender)
+        self.assert_json_error(result, "Notification type is required")
+
+    def test_typing_notification_nonprivate_type(self):
+        #type: () -> None
+        """
+        Sending typing notification with non-private type fails
+        """
+        sender = "hamlet@zulip.com"
+        reciever = "othelo@zulip.com"
+        result = self.client_post("/api/v1/typing", {"type": "private",
+                                                    "to": reciever},
+                                        **self.api_auth(sender)
+        self.assert_json_error(result, "Invalid typing notification type")
```

Comment:
For a lot of these tests, I would use `with tornado_redirected_to_list` to capture the events being issued and validate that the events have the expected content.